core/iwasm/interpreter/wasm_runtime.c

```diff
@@ -300,15 +296,15 @@ memory_instantiate(const WASMModule *module, WASMModuleInstance *parent,
                    uint8 *aux_heap_base_global_data, char *error_buf,
                    uint32 error_buf_size)
 {
+    bh_assert(memory != NULL);
```

Comment:
I thought we were sticking with declarations at the top of the block for compatibility with older C standards? This puts code before the declarations.

Reply:
Sounds good, I am for this new approach.